crates/turborepo-lib/src/task_hash.rs

```diff
@@ -242,6 +242,7 @@ pub struct TaskHasher<'a> {
     hashes: HashMap<TaskId<'static>, String>,
     run_opts: &'a RunOpts,
     env_at_execution_start: &'a EnvironmentVariableMap,
+    global_env: EnvironmentVariableMap,
```

Comment:
`global_env` previously lived inside of the `Visitor` structure. It is only used by `TaskHasher` so we keep it there instead of passing it in every time we construct a task's environment.